### PR TITLE
node Global view & account history

### PR DESCRIPTION
**Pull request by GuillaumeDSM**
requires https://github.com/Drakkar-Software/ccxt/pull/29
includes:
- global view update via scheduled job (ev. 5min)
- account history fetch & sync (stored in optimized sqlite) via daily scheduled job
- reduce tasks.db growing size by a lot

### CODE REVIEW

File: packages/flow/octobot_flow/repositories/exchange/transactions_repository.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import octobot_flow.repositories.exchange.base_exchange_repository as base_exchange_repository_import
+
+logger = commons_logging.get_logger("TransactionsRepository")
+
+
+class TransactionsRepository(base_exchange_repository_import.BaseExchangeRepository):
+
+    async def fetch_deposits(self, since: int = None, limit: int = None) -> list[dict]:
+        try:
+            return await self.exchange_manager.exchange.get_deposits(since=since, limit=limit)
```
Comment (Herklos) [inline packages/flow/octobot_flow/repositories/exchange/transactions_repository.py]: Could it make sense to create an updater the same way as trades and others and to use the updater here instead of the exchange manager function?

Comment (GuillaumeDSM) [reply packages/flow/octobot_flow/repositories/exchange/transactions_repository.py]: good point, it's not used but it would fit the architecture, i'll add an updater that for now will only be used there

**Review by Herklos — COMMENTED** [review]
Awesome testing framework! Great job

File: additional_tests/exchanges_tests/abstract_authenticated_exchange_tester.py [hunk 2]
```diff
@@ -776,6 +780,115 @@ async def inner_test_get_my_recent_trades(self):
...
+        ):
+            return True
+        return False
+
+    async def test_get_my_recent_trades_exhaust_history(self):
+        async with self.local_exchange_manager():
+            await self.inner_test_get_my_recent_trades_exhaust_history()
+
+    @expect_not_supported_on_read_only("get_my_recent_trades")
+    async def inner_test_get_my_recent_trades_exhaust_history(self):
```
Comment (Herklos) [inline additional_tests/exchanges_tests/abstract_authenticated_exchange_tester.py:815]: 👍

File: additional_tests/exchanges_tests/abstract_authenticated_exchange_tester.py [hunk 3]
```diff
@@ -776,6 +780,115 @@ async def inner_test_get_my_recent_trades(self):
...
+            assert len(exhaust) == len(single)
+            if exhaust:
+                self.check_raw_trades(exhaust)
+
+    async def test_get_deposits(self):
+        async with self.local_exchange_manager():
+            await self.inner_test_get_deposits()
+
+    @expect_not_supported_on_read_only("get_deposits")
+    async def inner_test_get_deposits(self):
```
Comment (Herklos) [inline additional_tests/exchanges_tests/abstract_authenticated_exchange_tester.py:839]: 👍

File: packages/tentacles/Services/Interfaces/node_api_interface/tests/test_routes_accounts.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        ):
+            response = tenant_client.get("/api/v1/accounts/acc-1/historical-values")
+        assert response.status_code == 404
+
+    def test_returns_503_when_scheduler_not_initialized(self, tenant_client, mock_auth):
+        with mock.patch("octobot_node.scheduler.is_initialized", return_value=False):
+            response = tenant_client.get("/api/v1/accounts/acc-1/historical-values")
+        assert response.status_code == 503
+
+    def test_returns_401_without_auth(self, client, mock_auth):
```
Comment (Herklos) [inline packages/tentacles/Services/Interfaces/node_api_interface/tests/test_routes_accounts.py:73]: 👍

File: packages/sync/octobot_sync/sync/collection_providers/user_account_history_provider.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+import octobot_commons.singleton.singleton_class as singleton_class
+import octobot_sync.constants as sync_constants
+import octobot_protocol.models as protocol_models
+import octobot_sync.enums as sync_enums
+
+import octobot_sync.sync.collection_backend.single_item_local_collection_provider as single_item_provider
+
+
+class AccountHistoryProvider(
```
Comment (Herklos) [inline packages/sync/octobot_sync/sync/collection_providers/user_account_history_provider.py:26]: 👍

File: packages/protocol/openapi.json [hunk 6]
```diff
@@ -818,6 +837,45 @@
           }
         }
       },
+      "TransactionType": {
```
Comment (Herklos) [inline packages/protocol/openapi.json:840]: 👍

File: packages/protocol/openapi.json [hunk 7]
```diff
@@ -1926,6 +1996,46 @@
...
+          },
+          "account_ids": {
+            "type": "array",
+            "items": {
+              "type": "string"
+            }
+          }
+        }
+      },
+      "ResetAccountTradingDataConfiguration": {
```
Comment (Herklos) [inline packages/protocol/openapi.json:2018]: 👍

File: packages/flow/octobot_flow/logic/portfolio_history/daily_price_cache_updater.py [hunk 8]
```diff
@@ -0,0 +1,387 @@
...
+        trading_api.merge_daily_prices_in_memory(daily_prices, fetch_symbol, closes_by_timestamp)
+
+
+def _parse_base_quote(symbol: str) -> tuple[str, str] | None:
+    if "/" not in symbol:
+        return None
+    base_asset, quote = symbol.split("/", 1)
+    if not base_asset or not quote or base_asset == quote:
+        return None
+    return base_asset, quote
```
Comment (Herklos) [inline packages/flow/octobot_flow/logic/portfolio_history/daily_price_cache_updater.py:109]: Shouldn't we use commons Symbol instead ?

File: packages/flow/octobot_flow/jobs/portfolio_history_job.py [hunk 9]
```diff
@@ -0,0 +1,375 @@
...
+
+def _filter_symbols_on_live_markets(
+    exchange_manager,
+    symbols: list[str],
+) -> list[str]:
+    live_symbols = set(exchange_manager.client_symbols or [])
+    return sorted(symbol for symbol in symbols if symbol in live_symbols)
+
+
+def _derive_price_symbols(
```
Comment (Herklos) [inline packages/flow/octobot_flow/jobs/portfolio_history_job.py:334]: Shouldn't we use commons.Symbol ?

File: packages/flow/octobot_flow/jobs/global_view_account_job.py [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+        )
+        global_view_persistence_module.persist_global_view_refresh_result(
+            self.user_id,
+            account.id,
+            refresh_result,
+            persist_open_orders=persist_open_orders,
+        )
+        # Update persisted latest-tickers cache from fetched ticker closes.
+        if exchange_refresh_result.ticker_closes:
+            await trading_api.update_latest_tickers(
```
Comment (Herklos) [inline packages/flow/octobot_flow/jobs/global_view_account_job.py:115]: 👍

**Review by Herklos — COMMENTED** [review]
Great job! And very nice testing work 💯

File: packages/copy/octobot_copy/validators/reference_account_outdated_orders.py [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+        if trader_order_type in (
+            trading_enums.TraderOrderType.BUY_MARKET,
+            trading_enums.TraderOrderType.SELL_MARKET,
+        ):
+            continue
+        replicable.append(order)
+    return replicable
+
+
+def is_order_impossible_at_market_price(
```
Comment (Herklos) [inline packages/copy/octobot_copy/validators/reference_account_outdated_orders.py:41]: 👍

File: packages/node/tests/scheduler/automations/test_automation_states_loader.py [hunk 12]
```diff
@@ -0,0 +1,422 @@
...
+        )
+
+        state_dict = automation_states_loader_module.get_automation_state_dict(workflow_status)
+
+        assert state_dict is not None
+        assert state_dict["automation"]["metadata"]["name"] == "from-output"
+
+
+class TestPatchTaskContentDegradedState:
+    def test_persists_degraded_state_in_task_content(self):
```
Comment (Herklos) [inline packages/node/tests/scheduler/automations/test_automation_states_loader.py:169]: 👍

File: packages/trading/octobot_trading/exchange_data/databases/market_data_sqlite_database.py [hunk 13]
```diff
@@ -0,0 +1,219 @@
...
+import octobot_trading.constants as trading_constants
+import octobot_trading.enums as trading_enums
+import octobot_trading.exchange_data.exchange_cache_key as exchange_cache_key_module
+import octobot_trading.exchange_data.prices.daily_prices_cache_types as daily_prices_cache_types
+
+MARKET_DATA_DB_FILENAME = "market_data.sqlite"
+SCHEMA_VERSION = 1
+
+
+class MarketDataSQLiteDatabase(base_sqlite_database.BaseSQLiteDatabase):
```
Comment (Herklos) [inline packages/trading/octobot_trading/exchange_data/databases/market_data_sqlite_database.py:17]: 👍

File: packages/trading/tests/personal_data/portfolios/history/test_history_from_trades_and_transaction_builder.py [hunk 14]
```diff
@@ -0,0 +1,345 @@
...
+
+        day_2_start = DAY_SECONDS * 2
+        assert result[day_2_start]["BTC"][TOTAL] == decimal.Decimal("0.5")
+
+        day_1_start = DAY_SECONDS
+        assert result[day_1_start]["BTC"][TOTAL] == decimal.Decimal("1")
+        assert result[day_1_start]["USDT"][TOTAL] == decimal.Decimal("0")
+
+
+class TestBuildHistoricalHoldingsDepositWithdrawal:
```
Comment (Herklos) [inline packages/trading/tests/personal_data/portfolios/history/test_history_from_trades_and_transaction_builder.py:125]: 👍

**Review by Herklos — COMMENTED** [review]
👍

**Review by Herklos — APPROVED** [review]
🚀
